applications/vanilla/controllers/class.discussionscontroller.php

```diff
@@ -722,40 +724,48 @@ public function tagged() {
         $Tag = stringEndsWith($Tag, '.rss', true, true);
         list($Offset, $Limit) = offsetLimit($Page, c('Vanilla.Discussions.PerPage', 30));
 
-        $MultipleTags = strpos($Tag, ',') !== false;
-
         $this->setData('Tag', $Tag, true);
 
-        $TagModel = TagModel::instance();
-        $RecordCount = false;
-        if (!$MultipleTags) {
-            $Tags = $TagModel->getWhere(array('Name' => $Tag))->resultArray();
+        $tagModel = TagModel::instance();
+        // If more than one tag has been passed via GET, pass them as an array to the model.
+        $multipleTags = strpos($Tag, ',') !== false;
+        if ($multipleTags) {
+            $tags = $tagModel->getWhere(['Name' => explode(',', $Tag)])->resultArray();
+        } else {
+            $tags = $tagModel->getWhere(['Name' => $Tag])->resultArray();
+        }
 
-            if (count($Tags) == 0) {
-                throw notFoundException('Page');
-            }
+        if (count($tags) == 0) {
+            throw notFoundException('Page');
+        }
 
-            if (count($Tags) > 1) {
-                foreach ($Tags as $TagRow) {
-                    if ($TagRow['CategoryID'] == val('CategoryID', $Category)) {
-                        break;
-                    }
+        // Get the number of discussions and the page title by either looping
+        // through all the tags or getting it from the result from the TagModel.
+        $recordCount = false;
+        if (count($tags) > 1) {
```

Comment:
We can probably get rid of this condition and foreach on tags no matter is there is one or more tags.